### PR TITLE
Add calib_astrometryUsed field to record sources used

### PR DESCRIPTION
**Pull request by pgee2000**
Also added a doWrite config field to match photoCal and measurePsf

### CODE REVIEW

**Comment by PaulPrice** [comment]
But that `Task` doesn't write anything, so the config field is a bit misleading.

File: python/lsst/meas/astrom/astrometry.py [hunk 1]
```diff
@@ -55,6 +57,11 @@ class AstrometryConfig(RefMatchConfig):
         default=0.001,
         min=0,
     )
+    doWriteOutput = pexConfig.Field(
```
Comment (parejkoj) [inline python/lsst/meas/astrom/astrometry.py]: I feel like a better name for this config parameter would be `doOutputUsedFlag`.

File: python/lsst/meas/astrom/astrometry.py [hunk 2]
```diff
@@ -147,6 +154,13 @@ def __init__(self, refObjLoader, schema=None, **kwargs):
         @param[in] kwargs  additional keyword arguments for pipe_base Task.\_\_init\_\_
         """
         RefMatchTask.__init__(self, refObjLoader, schema=schema, **kwargs)
+
+        if self.config.doWriteOutput and not schema is None:
```
Comment (parejkoj) [inline python/lsst/meas/astrom/astrometry.py]: `schema is not None`

File: python/lsst/meas/astrom/astrometry.py [hunk 3]
```diff
@@ -269,6 +283,9 @@ def solve(self, exposure, sourceCat):
             (iterNum, len(tryRes.matches), tryMatchDist.distMean.asArcseconds(),
                 tryMatchDist.distStdDev.asArcseconds()))
 
+        for m in res.matches:
+            if self.usedKey:
```
Comment (parejkoj) [inline python/lsst/meas/astrom/astrometry.py]: probably should be explicit here: `if self.usedKey is not None`

File: python/lsst/meas/astrom/astrometry.py [hunk 4]
```diff
@@ -147,6 +154,13 @@ def __init__(self, refObjLoader, schema=None, **kwargs):
         @param[in] kwargs  additional keyword arguments for pipe_base Task.\_\_init\_\_
         """
         RefMatchTask.__init__(self, refObjLoader, schema=schema, **kwargs)
+
+        if self.config.doWriteOutput and not schema is None:
+            self.usedKey = schema.addField("calib_astrometryUsed", type="Flag",
+                                               doc="set if source was used in astrometric calibration")
```
Comment (parejkoj) [inline python/lsst/meas/astrom/astrometry.py]: funky indentation

**Comment by parejkoj** [comment]
And we need a unittest of this functionality here as well.

File: python/lsst/meas/astrom/astrometry.py (hunk 1 above)
Comment (pgee2000) [inline python/lsst/meas/astrom/astrometry.py]: I would like to leave this consistent with the other calib modules.

File: python/lsst/meas/astrom/astrometry.py (hunk 2 above)
Comment (pgee2000) [inline python/lsst/meas/astrom/astrometry.py]: I'm not a fan of this code standard, but OK.

File: python/lsst/meas/astrom/astrometry.py (hunk 4 above)
Comment (pgee2000) [inline python/lsst/meas/astrom/astrometry.py]: OK

File: python/lsst/meas/astrom/astrometry.py (hunk 3 above)
Comment (pgee2000) [inline python/lsst/meas/astrom/astrometry.py]: OK

**Comment by pgee2000** [comment]
I have not bothered to add a unit test here.  I don't think it is worth the bother, and writing any kind of intelligent unit test requires an understanding of the Task, which is well beyond the scope of this ticket.  I did add one to photoCal, where it makes more sense.

File: python/lsst/meas/astrom/astrometry.py (hunk 1 above)
Comment (PaulPrice) [inline python/lsst/meas/astrom/astrometry.py]: I agree that this should be changed. This `Task` does not do any writing, so the name is misleading.

**Comment by PaulPrice** [comment]
It looks like you've merged prematurely.

File: python/lsst/meas/astrom/astrometry.py [hunk 5]
```diff
@@ -22,6 +22,8 @@
 from __future__ import absolute_import, division, print_function
 from builtins import range
 
+import random
+
```
Comment (laurenam) [inline python/lsst/meas/astrom/astrometry.py]: It doesn't look like this is used.

**Pull request merged**